wsmaster/che-core-api-workspace/src/main/resources/schema/devfile.json

```diff
@@ -162,20 +162,52 @@
                 }
               },
               "then": {
-                "additionalProperties": true,
-                "required": [
-                  "id"
+                "oneOf": [
+                  {
+                    "required": [
+                      "id"
+                    ],
+                    "not": {
+                      "required": [
+                        "reference"
+                      ]
+                    }
+                  },
+                  {
+                    "required": [
+                      "reference"
+                    ],
+                    "not": {
+                      "required": [
+                        "id"
+                      ]
+                    }
+                  }
                 ],
                 "properties": {
                   "type": {},
                   "alias": {},
                   "id": {
                     "type": "string",
-                    "description": "Describes the component id. It has the following format: [{REGISTRY_URL}/]{plugin/editor PUBLISHER}/{plugin/editor NAME}/{plugin/editor VERSION}, where {REGISTRY_URL}/ is an optional part.",
-                    "pattern": "^((https?://)[a-zA-Z0-9_\\-./]+/)?[a-z0-9_\\-.]+/[a-z0-9_\\-.]+/[a-z0-9_\\-.]+$",
+                    "description": "Describes the component id. It has the following format: {plugin/editor PUBLISHER}/{plugin/editor NAME}/{plugin/editor VERSION}",
+                    "pattern": "[a-z0-9_\\-.]+/[a-z0-9_\\-.]+/[a-z0-9_\\-.]+$",
+                    "examples": [
+                      "eclipse/maven-jdk8/1.0.0"
+                    ]
+                  },
+                  "reference": {
+                    "description": "Describes raw location of plugin yaml file.",
+                    "type": "string",
+                    "examples": [
+                      "https://pastebin.com/raw/kYprWiNB"
+                    ]
+                  },
+                  "registryUrl": {
+                    "description": "Describes URL of custom plugin registry.",
+                    "type": "string",
+                    "pattern": "^((https?://)[a-zA-Z0-9_\\-./]+)",
```

Comment:
It's the same as the following examples:
1. `^(https?://)[a-zA-Z0-9_\\-./]+`
2. `^https?://[a-zA-Z0-9_\\-./]+`

I prefer first way